doc/contributing.rst

```diff
@@ -18,47 +18,69 @@ See `codeowners <codeowners.html>`_ for more details.
 Writing tests
 =============
 
-There are 3 types of tests:
+There are 4 types of tests:
 
 1. ``runnableExamples`` documentation comment tests, ran by ``nim doc mymod.nim``
    These end up in documentation and ensure documentation stays in sync with code.
 
-2. tests in ``when isMainModule:`` block, ran by ``nim c mymod.nim``
-   ``nimble test`` also typially runs these in external nimble packages.
+2. separate test files, e.g.: ``tests/stdlib/tos.nim``.
+   In nim repo, `testament` (see below) runs all `$nim/tests/*/t*.nim` test files;
+   in nimble packages, `nimble test` by default runs all `$pkg/tests/t*.nim`
+   test files when no `task "test"` is specified.
+   (see https://github.com/nim-lang/nimble#tests for latest instructions).
 
-3. testament tests, e.g.: ``tests/stdlib/tos.nim`` (only used for Nim repo).
+3. (not preferred) tests in ``when isMainModule:`` block, ran by ``nim r mymod.nim``.
```

Comment:
I think that `when isMainModule:` on individual modules should start its Deprecation way on stdlib,
and move it into `runnableExamples` and/or Testament tests, Nim has better and faster ways for tests now,
...so it should be documented as a Deprecated.

Reply:
done